pywbem/cim_operations.py

```diff
@@ -1709,6 +1741,299 @@ def OpenEnumerateInstancePaths(self, ClassName, namespace=None,
+            if self.use_pull_operations is None or self.use_pull_operations:
+                try:        # operation try block
+                    pull_result = self.OpenEnumerateInstances(
+                        ClassName, namespace=namespace, LocalOnly=LocalOnly,
+                        DeepInheritance=DeepInheritance,
+                        IncludeQualifiers=IncludeQualifiers,
+                        IncludeClassOrigin=IncludeClassOrigin,
+                        PropertyList=PropertyList,
+                        FilterQueryLanguage=FilterQueryLanguage,
+                        FilterQuery=FilterQuery,
+                        OperationTimeout=OperationTimeout,
+                        ContinueOnError=ContinueOnError,
+                        MaxObjectCount=MaxObjectCount, **extra)
+
+                    # Open operation succeeded; set has_pull flag
+                    self.use_pull_operations = True
+
+                    for inst in pull_result.instances:
+                        yield inst
+
+                    # loop to pull while more while eos not returned.
+                    while not pull_result.eos:
+                        pull_result = self.PullInstancesWithPath(
+                            pull_result.context, MaxObjectCount=MaxObjectCount)
+
+                        for inst in pull_result.instances:
+                            yield inst
+                    pull_result = None   # clear the pull_result
+                    return
+
+                # If NOT_SUPPORTED and first request, set flag and try
+                # alternative request operation.
+                # If use_pull_operations is True, always raise the exception
+                except CIMError as ce:
+                    if self.use_pull_operations is None and \
+                       ce.status_code == CIM_ERR_NOT_SUPPORTED:
+                        self.use_pull_operations = False
+                    else:
+                        raise
+
+            # Alternate request if Pull not implemented. This does not allow
+            # the FilterQuery or ContinueOnError
+            assert self.use_pull_operations is False
+
+            if FilterQuery is not None or FilterQueryLanguage is not None:
+                raise ValueError('EnumerateInstance does not support Filters.')
+
+            if ContinueOnError is not None:
+                raise ValueError('EnumerateInstance does not support '
+                                 'ContinueOnError.')
+
+            enum_rslt = self.EnumerateInstances(
+                ClassName, namespace=namespace, LocalOnly=LocalOnly,
+                DeepInheritance=DeepInheritance,
+                IncludeQualifiers=IncludeQualifiers,
+                IncludeClassOrigin=IncludeClassOrigin,
+                PropertyList=PropertyList, **extra)
+
+            for inst in enum_rslt:
+                yield inst
+
+        # Clean up if close() executed and  pull operation is active
```

Comment:
I finally understand the situation this comment describes: The user is in the loop iterating over the items yielded by this method (i.e. iterates over the returned iterator), and for some reason calls close() on the returned iterator before the iterator is exhausted. Using the wording "and pull operation is active" is a bit confusing to me, because it is not the pull operation against the server that is active, but the iterator is not yet exhausted.

Can we change this comment to:

```
# Clean up if the caller closes the iterator before exhausting it
```